### ISSUE TITLE
support for JSON Binding API (JSON-B)

### ISSUE THREAD

**Issue by cvgaviao** [immutables/immutables#1115]
It would be nice to have the possibility to use my generated immutables classes with a [json-B](http://json-b.net/) engine easily.

Json-b current spec lacks an out-out-box annotated polymorphic support, but can be achieved with custom JsonbDeserializer class. But a spec update containing it is being baked and Yasson and Johnzon projects already started to implement it.

**Comment by asereda-gs** [comment]
This was already requested (see #708). 

Adding support of new JSON API (library) in immutables is quite involved. Before starting working on it we would like to see some signs that it is being actively used by community. 

Also internally we're also discussing about marshaling / unmarshaling abstractions. There is no just JSON but also binary formats like protobuf, BSON, Flatbuffers, Kryo etc. I think long-term this will pay-off.
